### PR TITLE
Logout

### PR DESCRIPTION
**Pull request by rmai99**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

## Type of change

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

### CODE REVIEW

File: services/base-http-client/index.ts [hunk 1]
```diff
@@ -13,10 +13,12 @@ type Config = {
 };
 
 // TODO
-// interface ErrorResponse {
-//   statusCode: string;
-//   statusDescription: string;
-// }
+interface ErrorResponse {
```
Comment (husniramdani) [inline services/base-http-client/index.ts:16]: is there no error code?

File: services/base-http-client/index.ts [hunk 2]
```diff
@@ -13,10 +13,12 @@ type Config = {
 };
 
 // TODO
-// interface ErrorResponse {
```
Comment (husniramdani) [inline services/base-http-client/index.ts:16]: don't hesitate to delete it when u already solved it

File: services/base-http-client/index.ts [hunk 3]
```diff
@@ -111,6 +113,12 @@ export abstract class BaseHttpClient {
         }
       },
       (error: AxiosError) => {
+        const responseData = error?.response?.data as ErrorResponse;
+        if (responseData.message == 'invalid or expired jwt') {
```
Comment (husniramdani) [inline services/base-http-client/index.ts:117]: better use triple equation ===

Comment (husniramdani) [reply services/base-http-client/index.ts:117]: is there no specific error code for expired jwt request?

File: services/user-service/index.ts [hunk 4]
```diff
@@ -35,4 +35,7 @@ export default class UserService extends BaseHttpClient {
   getListRole(): Promise<Array<TDropdown>> {
     return this.get('/list-role');
   }
+  requestLogoutUser() {
```
Comment (husniramdani) [inline services/user-service/index.ts:38]: LGTM! already follows the name convention. but it is better to place it under the login request

File: src/layouts/default.vue [hunk 5]
```diff
@@ -179,7 +179,9 @@ export default defineComponent({
     },
     // TODO : create confirmation modal logout
```
Comment (husniramdani) [inline src/layouts/default.vue:180]: don't forget to delete the todo when u done with the task

File: services/base-http-client/index.ts (hunk 3 above)
Comment (rmai99) [reply services/base-http-client/index.ts:117]: no. from jwt lib it return code 401 and message invalid or expired jwt

**Comment by husniramdani** [comment]
LGTM!

**Pull request merged**